### PR TITLE
M3-268 nodebalancer summary

### PR DESCRIPTION
**Pull request by martinmckenna**
### Purpose

Add nodebalancer summary information
* contains basic info about the nodebalancer
* contains connections/second and incoming and outcoming traffic/second graphs

### Notes

* Created a new LineGraph component
* No errors appear for bad requests to `nodebalancer/stats`, as per review notes on Linode stats errors
* No filtering yet

### CODE REVIEW

**Comment by martinmckenna** [comment]
@alioso this will need a bit of work. I did my best to copy over styles from LinodeSummary, but unfortuantely wasn't able to cover all the bases

File: src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerSummary/NodeBalancerSummary.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+interface Props {
+  nodeBalancer: Linode.ExtendedNodeBalancer;
+}
+
+interface State { }
+
+type CombinedProps = Props & WithStyles<ClassNames>;
+
+class NodeBalancerSummary extends React.Component<CombinedProps, State> {
```
Comment (martinmckenna) [inline src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerSummary/NodeBalancerSummary.tsx]: make this a stateless component

**Comment by alioso** [comment]
@rjstires styling done for this one. One thing that differs from the mockup is the region flag (not displaying currently as we are reusing the component from LinodesDetail which does not have it. I'd say we either have it on both or not whatsoever.

@martinmckenna we do need a rebase here

File: src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerSummary/SummaryPanel.tsx [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+            variant="caption"
+          >
+            <strong>Host Name:</strong> {nodeBalancer.hostname}
+          </Typography>
+        </Grid>
+        <Grid item xs={12} sm={6}>
+          <Typography variant="caption">
+            <strong>
+              Ports: </strong> {nodeBalancer.ports.length === 0 && 'None'}
+              {nodeBalancer.ports.map((port, index, ports) => {
```
Comment (asauber) [inline src/features/NodeBalancers/NodeBalancerDetail/NodeBalancerSummary/SummaryPanel.tsx]: You can simply use `nodeBalancer.ports.join(', ')`

File: src/components/LineGraph/LineGraph.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+  showToday: boolean;
+  suggestedMax?: number;
+  data: DataSet[];
+}
+
+interface State { }
+
+type CombinedProps = Props & WithStyles<ClassNames>;
+
+const chartOptions: any = {
```
Comment (asauber) [inline src/components/LineGraph/LineGraph.tsx]: We don't want these `chartOptions` to diverge from those used on LinodeSummary, so this new component should be re-used there.

**Comment by martinmckenna** [comment]
@asauber @rjstires ready for re-review

It felt good to delete all those lines from LinodeSummary

**Comment by asauber** [comment]
screen shot 2018-05-30 at 3 12 31 pm

Echoing @rjstires, we should show the error state component instead of "Graphs" with blank space below it when the stats call fails.

Also @alioso the placement of the items in the summary panel doesn't look right.

**Pull request merged**